hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/DelegatingFileSystem.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.crypto2.hadoop;
+
+import java.io.IOException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.FilterFileSystem;
+import org.apache.hadoop.fs.Path;
+
+/**
+ * Equivalent to {@link FilterFileSystem} but invokes {@link #create} and {@link #open} on this class when calling
+ * {@link #copyFromLocalFile} and {@link #copyToLocalFile}.
+ * <p>
+ * Solves: https://issues.apache.org/jira/browse/HADOOP-13870
+ */
+public abstract class DelegatingFileSystem extends FilterFileSystem {
+
+    public DelegatingFileSystem(FileSystem delegate) {
+        super(delegate);
+        try {
+            super.initialize(delegate.getUri(), delegate.getConf());
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to initialize the delegating filesystem", e);
+        }
+    }
+
+    public void copyFromLocalFile(Path src, Path dst) throws IOException {
```

Comment:
no override? should add javadoc in that case

Reply:
done